Refactors Edit Access modal and adds app-specific warnings

jadmnsaad opened the pull request:
Summary
- Refactors the Edit Access modal to extract action logic and UI pieces into separate modules.
- Adds app-specific access level warning messages and centralizes modal options/constants.

Why
- Improves separation of concerns by moving access update/revoke/deny logic into a reusable hook.
- Simplifies the modal component by extracting selection and warning UI into their own components.
- Provides clearer, app-specific guidance to admins when selecting certain access levels.
- Consolidates error handling, loading state, and notifications for more consistent behavior.

What changed
- Introduces a useAccessActions hook that encapsulates revoke, deny, and update access flows, error handling, and loading state.
- Adds AccessLevelSelection and AccessLevelWarning components to render the radio options and contextual informational alerts.
- Adds constants for group paths, action types, modal option labels, and access-level warning text.
- Simplifies the modal component to rely on the new hook and components; updates how the modal closes and refetches user data after successful actions.
- Improves UI layout in several places (e.g., replaces inline div wrappers with Box, moves loading state into hook).

Notes
- Preserves previous behavior while making the code easier to maintain and extend (e.g., adding new app-specific warnings).
- Relates to CENTRE-task#59

Checklist
- [x] Refactor logic into reusable hook
- [x] Extract UI subcomponents
- [x] Add app-specific warning content

Commit:
Adds app-specific access level warnings

Implements app-specific warning messages within the Edit Access modal to provide users with additional context when assigning roles.
These messages inform users about potential implications of assigning specific access levels, such as EAO Team Member in EPIC.engage or Viewer in EPIC.track, and guide them to the appropriate app management section for assigning users to engagements or works.
Removes the generic access level warning.

Related to CENTRE-task#59

Commit:
Refactors edit access modal

Improves the structure of the edit access modal by introducing reusable components for access level selection and warnings.

Moves access action logic into a dedicated hook for better separation of concerns and testability.

This change enhances code maintainability and readability.

Commit:
Allow only DST admins to edit condition repo access

saravanpa-aot commented on centre-api/src/centre_api/enums/epic_app.py:
```diff
@@ -102,6 +102,6 @@ class EpicAdminSubGroups(Enum):
     EpicAppClientName.EPIC_ENGAGE.value: ['create_admin_user'],
     EpicAppClientName.EPIC_COMPLIANCE.value: ['super_user'],
-    EpicAppClientName.CONDITION_REPOSITORY.value: ['view_conditions'],
+    EpicAppClientName.CONDITION_REPOSITORY.value: [''],
```
this is not needed?
con repo has this role to check for access

jadmnsaad replied:
There are no admin roles in Condition Repo. Nothing equivalent of super users.

<img width="1140" height="276" alt="Image" src="https://github.com/user-attachments/assets/088473c3-d9f5-41d9-870f-56e5fc7a9467" />

The role is called Admin but it's like administrating condition repo operations, it's not a manage auth role

saravanpa-aot commented on centre-web/src/components/AuthManagement/EditAccess/util.ts:
```diff
@@ -0,0 +1,11 @@
+export const getAccessLevelWarningMessage = (groupPath: string) => {
+  if (groupPath === "/ENGAGE/EAO_TEAM_MEMBER") {
```
may be make the group names a constant somewhere?

jadmnsaad replied:
Good eye, done

Commit:
Centralizes Keycloak group definitions

Introduces a new `KCGroup` model to consolidate all Keycloak group paths and metadata into a single, immutable `EPIC_GROUPS` constant. This eliminates hardcoded group paths throughout the application, enhancing consistency and maintainability. Updates existing components to leverage this centralized source of truth.

Relates to CENTRE-task#59

Commit:
Refines DeepReadonly function type detection

Updates the `DeepReadonly` utility type to use a more specific type signature for functions. This ensures only callable functions are correctly identified and excluded from deep readonly transformation, preventing potential issues with the broader `Function` type matching.

Relates to CENTRE-task#59
